### PR TITLE
Add prune to child scene layers

### PR DESCRIPTION
**Pull request by abarth**
Also, change the hit testing geometry so that we get our hits at the root.

### CODE REVIEW

**Comment by abarth** [comment]
@j9brown

File: flow/layers/child_scene_layer.h [hunk 1]
```diff
@@ -27,16 +27,19 @@ class ChildSceneLayer : public Layer {
 
   void set_scene_token(uint32_t scene_token) { scene_token_ = scene_token; }
 
+  void set_prune(bool prune) { prune_ = prune; }
```
Comment (chinmaygarde) [inline flow/layers/child_scene_layer.h]: I think we can do a better job of naming this method. It took me a while to make the connection that `set_prune` on the `ChildSceneLayer` was actually referring to the layers hit-test behavior (and only after opening up the `.fidl` file and the `.cc`). Maybe `set_hit_test_behavior` or something more indicative of this setter controlling hit-testing specifically?

Also, I am not sure I understand the relationship between `HitTestBehavior.Visibility.INVISIBLE` and `prune`.

Comment (abarth) [inline flow/layers/child_scene_layer.h]: Jeff suggested "hittable".  The difference between this flag and "invisible" is that it prevents children from being hit.  Invisible just prevents this node specifically from being hit.

Comment (chinmaygarde) [inline flow/layers/child_scene_layer.h]: > The difference between this flag and "invisible" is that it prevents children from being hit. Invisible just prevents this node specifically from being hit.

Thanks for clearing that up. Assumed that sublayers of invisible layers would be ineligible for hit-testing by default.

**Comment by tvolkert** [comment]
@abarth it looks like this causes the engine buildbots to [fail](https://build.chromium.org/p/client.flutter/builders/Mac%20Engine/builds/989/steps/steps/logs/stdio)

```
Analyzing dart:ui library...
[error] Expected an identifier (/b/build/slave/Mac_Engine/build/src/out/host_debug_unopt/gen/sky/bindings/dart_ui/compositing.dart, line 201, col 3)
[lint] Name non-constant identifiers using lowerCamelCase. (/b/build/slave/Mac_Engine/build/src/out/host_debug_unopt/gen/sky/bindings/dart_ui/compositing.dart, line 201, col 3)
step returned non-zero exit code: 1
```

**Comment by abarth** [comment]
Perhaps the engine buildbot is using an older analyzer that doesn't understand trailing commas.  I'll try to fix it.

**Comment by abarth** [comment]
That appears to have healed it.  Thanks!

**Pull request merged**